spec/term/1.0.md

```diff
@@ -0,0 +1,204 @@
+If it does so, it SHOULD discard entire lines or paragraphs, that is, the last codepoint discarded SHOULD be U+000A, U+000C, U+2028 or U+2029.
+
+### 1.3. Standard input
+
+TODO describe how keypresses are reflected on stdin (canonical/raw mode, noecho property, etc.), how Ctrl-D is handled
+
+### 1.4. Display
+
+If the server displays the document to a user, or renders it to some graphics device or file, then the terminal document SHALL be laid out according to the rules for normalization, writing direction, line breaking and handling of bidirectional text as defined in the Unicode Standard, as published by the [Unicode consortium](https://www.unicode.org).
+
+The surface onto which the terminal document is rendered is called the **terminal viewport**.
+
+### 1.5. Scrollback
+
+The server MAY impose limits on the size of the terminal document (e.g. because of memory constraints).
+When new text is added to the terminal document in such a way that the resulting terminal document would exceed those limits, the server MAY discard a prefix of the terminal document.
+If the cursor points into that prefix or at its start, the cursor SHALL be moved to the end of the prefix before the prefix is discarded.
+
+The server SHOULD choose the prefix such that its end is a line separator, paragraph separator, or a line break control character (U+0010).
+
+*Rationale:* In other words, servers should prefer to discard whole lines at a time.
+Chomping off only parts of a line causes unnecessary confusion on the side of the user.
+
+## 2. Message types for `vt6/term1`
+
+### 2.1. The `term.make-stdio` message
+
+- Directionality: client to server
+- Number of arguments: zero
+
+A client can send this message to turn this server connection into a standard input/output.
+On success, the server answers with `term.is-stdio`, see below.
+
+### 2.2. The `term.is-stdio` message
+
+- Directionality: server to client
+- Number of arguments: zero
+
+A server sends this message as a successful response to a `term.make-stdio` message.
+After this message, this server connection ceases to be used as a server connection and can be used as a standard input/output instead:
+
+TODO explain in detail (how stdin works, how stdout works, how anything that might send a server-initiated message is cancelled)
+
+TODO maybe remove the notion of standard input/output from core1.0 entirely and introduce it here?
+
+## 3. Properties for `vt6/term1`
+
+TODO define connection-lifetime-scoped properties
+
+### 3.1. The `term.width` property
+
+- Acceptable values: unsigned integers
+- Can be set by client: no
+
+The value of this property is the maximum number of printable ASCII characters (Unicode codepoints U+0020-U+007F) that can be displayed on the terminal viewport without having to break the line or overflow the viewport.
```

Comment:
I don't see how this makes any sense outside of mono.

Reply:
OK.